FWCore/Modules/src/LogErrorHarvester.cc

```diff
@@ -41,10 +46,42 @@ namespace edm {
     void beginJob() override;
     void produce(StreamID, Event&, EventSetup const&) const override;
     void endJob() override;
+    
   };
 
-  LogErrorHarvester::LogErrorHarvester(ParameterSet const&) {
+  LogErrorHarvester::LogErrorHarvester(ParameterSet const& iPSet) {
+    
     produces<std::vector<ErrorSummaryEntry>>();
+
+    const edm::TypeID endPathStatusType{typeid(edm::EndPathStatus)};
+    const edm::TypeID pathStatusType{typeid(edm::PathStatus)};
+    const edm::TypeID triggerResultsType{typeid(edm::TriggerResults)};
+
+    auto const& ignore = iPSet.getUntrackedParameter<std::vector<std::string>>("excludeModules");
+    const std::unordered_set<std::string> excludedModules(ignore.begin(),ignore.end());
+
+    auto const& includeM = iPSet.getUntrackedParameter<std::vector<std::string>>("includeModules");
+    const std::unordered_set<std::string> includeModules(includeM.begin(),includeM.end());
+
+    //Need to be sure to run only after all other EDProducers have run
+    callWhenNewProductsRegistered([this,
+                                   endPathStatusType,pathStatusType,triggerResultsType,
+                                   excludedModules, includeModules](edm::BranchDescription const& iBD) 
+    {
+      if((iBD.branchType() == edm::InEvent and moduleDescription().processName() == iBD.processName()) and 
+         ( (includeModules.empty() or
+            includeModules.end() != includeModules.find(iBD.moduleLabel())) and
+           (iBD.unwrappedTypeID() != endPathStatusType and
+            iBD.unwrappedTypeID() != pathStatusType and
+            iBD.unwrappedTypeID() != triggerResultsType))) {
+        if(excludedModules.end() == excludedModules.find(iBD.moduleLabel())) {
+          consumes(edm::TypeToGet{iBD.unwrappedTypeID(),edm::PRODUCT_TYPE},
+                   edm::InputTag{iBD.moduleLabel(),
+                                 iBD.productInstanceName(),
+                                 iBD.processName()});
```

Comment:
This is a huge showstopper. We need to enhance earlyDelete and other mechanisms to reduce memory and improve concurrency